keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,995 @@
+}
+```
+
+### Resource Claim API
+
+A special resource claim object is created to keep track of device allocations for
+extended resource. The resource claim object has following properties:
+
+  * It is namespace scoped, like other resource claim objects.
+  * It is owned by a pod, like other resource claim objects.
+  * It has `spec` of devices.requests, with each request name being a hash of the
+    container name and the extended resource backed by DRA name inside the container.
+  * Its `status.allocation.devices` and `status.allocation.reservedFor` are
+    used.
+  * It does not have annotation `resource.kubernetes.io/pod-claim-name:` as
+    it is created for the extended resource request in a pod spec, not for a
+    claim in the pod spec.
+  * At most one such claim object is created per pod. For example, if the pod
+    requests for foo1.domain/bar and foo2.domain/bar, the allocation of devices
+    for each are recorded in DeviceResourceRequestAllocationResult, and
+    one claim object with two `Results` is created for the pod.
+
+The special resource claim object lifecycle is managed by the scheduler and
+garbage collector.
+
+  * It is *created* in a namespace when there is a pod with extended resource
+  request, and the extended resource is advertised by `ResourceSlice` and
+  scheduler has fit the pod to a node with the `ResourceSlice`.
+  * It is *created* by the scheduler dynamic resource plugin during
+    preBind phase. The in-memory one in the assumed cache is created earlier
+    during Reserve phase.
+  * It is *deleted*
+    * either together with the owning pod's deletion.
+    * or by the scheduler dynamic resource plugin during unReserve phase.
+  * It is *read* by the kubelet DRA device driver to prepare the devices listed
+    therein when preparing to run the pod.
+
+```go
+type DeviceRequest struct {
+	// Name can be used to reference this request in a pod.spec.containers[].resources.claims
+	// entry and in a constraint of the claim.
+	//
+	// Must be a DNS label.
+	//
+	// +required
+	Name string `json:"name" protobuf:"bytes,1,name=name"`
+}
+```
+
+As shown above, device request must have a name, and it must be a DNS label.
+kube-scheduler creates one `DeviceRequest` per extended resource backed by DRA
+per container in the pod. The name of the `DeviceRequest` is determined by the
+container name, and the extended resource backed by DRA name. It is in the
+form of "c%d-e%d", where the first %d is the index of the container in the pod.
+The second %d is the index of the extended resource inside the container
+resource requests. For example, if the first container in the pod has an
+extended resource backed by DRA which is the 3rd such request in the container,
+then the name of the `DeviceRequest` is "c0-e2".
+
+### Pod API
+
+A new field `extendedResourceClaimStatus` is added to Pod's status to track
```

Comment:
Looks like you arrived at a similar conclusion as I did above... good! :smile: 

Let's make sure that this is called out above, otherwise readers will be kept scratching their head about how this will work for too long.

Reply:
Yes, added `extendedResourceClaimStatus` earlier in the doc such that readers can get the basic idea.